internal/middleware/compression.go

```diff
@@ -18,27 +18,63 @@ import (
 	"bytes"
 	"compress/gzip"
 	"compress/zlib"
+	"fmt"
 	"io"
 	"net/http"
+
+	"github.com/golang/snappy"
+	"github.com/klauspost/compress/zstd"
 )
 
 const (
 	headerContentEncoding = "Content-Encoding"
-	headerValueGZIP       = "gzip"
+	CompressionGzip       = "gzip"
+	CompressionZlib       = "zlib"
+	CompressionDeflate    = "deflate"
+	CompressionSnappy     = "snappy"
+	CompressionZstd       = "zstd"
 )
 
 type CompressRoundTripper struct {
-	http.RoundTripper
+	RoundTripper    http.RoundTripper
+	CompressionType string
+	Writer          func(*bytes.Buffer) (io.WriteCloser, error)
```

Comment:
Do these fields need to be exported?  It seems unwise to allow them to be changed after creation.

```suggestion
	compressionType string
	writer          func(*bytes.Buffer) (io.WriteCloser, error)
```

Reply:
done